dlt-connector/src/controller/TransactionBody.test.ts

```diff
@@ -0,0 +1,162 @@
+import 'reflect-metadata'
+import { TransactionDraft } from '@/graphql/input/TransactionDraft'
+import { create, determineCrossGroupType, determineOtherGroup } from './TransactionBody'
+import { UserIdentifier } from '@/graphql/input/UserIdentifier'
+import { TransactionError } from '@/graphql/model/TransactionError'
+import { TransactionErrorType } from '@/graphql/enum/TransactionErrorType'
+import { CrossGroupType } from '@/graphql/enum/CrossGroupType'
+import { TransactionType } from '@/graphql/enum/TransactionType'
+import Decimal from 'decimal.js-light'
+
+describe('test controller/TransactionBody', () => {
+  describe('test create ', () => {
+    const senderUser = new UserIdentifier()
+    const recipientUser = new UserIdentifier()
+    it('test with contribution transaction', () => {
+      const transactionDraft = new TransactionDraft()
+      transactionDraft.senderUser = senderUser
+      transactionDraft.recipientUser = recipientUser
+      transactionDraft.type = TransactionType.CREATION
+      transactionDraft.amount = new Decimal(1000)
+      transactionDraft.createdAt = '2022-01-02T19:10:34.121'
+      transactionDraft.targetDate = '2021-12-01T10:05:00.191'
+      const body = create(transactionDraft)
+
+      expect(body.creation).toBeDefined()
+      expect(body).toMatchObject({
+        createdAt: {
+          seconds: 1641150634,
+          nanoSeconds: 121000000,
+        },
+        versionNumber: '3.3',
+        type: CrossGroupType.LOCAL,
+        otherGroup: '',
+        creation: {
+          recipient: {
+            amount: '1000',
+          },
+          targetDate: {
+            seconds: 1638353100,
+          },
+        },
+      })
+    })
+    it('test with local send transaction send part', () => {
+      const transactionDraft = new TransactionDraft()
+      transactionDraft.senderUser = senderUser
+      transactionDraft.recipientUser = recipientUser
+      transactionDraft.type = TransactionType.SEND
+      transactionDraft.amount = new Decimal(1000)
+      transactionDraft.createdAt = '2022-01-02T19:10:34.121'
+      const body = create(transactionDraft)
+
+      expect(body.transfer).toBeDefined()
+      expect(body).toMatchObject({
+        createdAt: {
+          seconds: 1641150634,
+          nanoSeconds: 121000000,
+        },
+        versionNumber: '3.3',
+        type: CrossGroupType.LOCAL,
+        otherGroup: '',
+        transfer: {
+          sender: {
+            amount: '1000',
+          },
+        },
+      })
+    })
+
+    it('test with local send transaction receive part', () => {
+      const transactionDraft = new TransactionDraft()
+      transactionDraft.senderUser = senderUser
+      transactionDraft.recipientUser = recipientUser
```

Comment:
I definitely need to keep that in mind, but I'm not quite done yet and need to figure out at this point how exactly I'm going to do that. In my model there is only a SEND and a RECEIVE transaction for cross-group transactions and they are identical except for the otherGroup field of TransactionBody. With LOCAL Transactions there is in principle only the SEND Transaction, because the RECEIVE Transaction would be identical and I try to be storage saving.